File: frontend/src/types/configmap.ts
export interface ConfigMapSummary {
  name: string;
  namespace: string;
  data_keys: string[];
  created_at: string | null;
}

export interface ConfigMapDetail {
  name: string;
  namespace: string;
  data: Record<string, string>;
  created_at: string | null;
}

File: backend/app/schemas/configmap.py
from datetime import datetime

from pydantic import BaseModel, Field


class ConfigMapSummary(BaseModel):
    name: str
    namespace: str
    data_keys: list[str]
    created_at: datetime | None


class ConfigMapDetail(BaseModel):
    name: str
    namespace: str
    data: dict[str, str]
    created_at: datetime | None


class ConfigMapCreate(BaseModel):
    name: str = Field(..., pattern=r"^[a-z][a-z0-9-]{0,62}$")
    namespace: str = "pi-apps"
    data: dict[str, str] = Field(default_factory=dict)


class ConfigMapUpdate(BaseModel):
    data: dict[str, str]
